tests/conftest.py

```diff
@@ -577,33 +577,36 @@ def pytest_runtest_setup(item):
     ):
         item._skipped_by_mark = True
         pytest.skip(PRE_PYTEST_SKIP_REASON)
+    test_group_count = sum(
+        bool(item.get_closest_marker(group)) for group in ("core_test", "slow_test")
+    )
+    if item.get_closest_marker("core_test") and item.get_closest_marker("slow_test"):
+        raise pytest.UsageError(
+            "Tests can only be in one test group. ('core_test', 'slow_test')"
+        )
 
-    if item.get_closest_marker("core_test"):
-        if not item.config.getoption("--core-tests"):
-            raise pytest.skip.Exception(
-                "Core tests are disabled, pass '--core-tests' to enable them.",
-                _use_item_location=True,
-            )
-    if item.get_closest_marker("slow_test"):
-        if not item.config.getoption("--slow-tests"):
-            raise pytest.skip.Exception(
-                "Slow tests are disabled, pass '--run-slow' to enable them.",
-                _use_item_location=True,
-            )
     if item.get_closest_marker("flaky_jail"):
         if not item.config.getoption("--flaky-jail"):
             raise pytest.skip.Exception(
                 "flaky jail tests are disabled, pass '--flaky-jail' to enable them.",
                 _use_item_location=True,
             )
-    if (
-        not item.get_closest_marker("slow_test")
-        and not item.get_closest_marker("core_test")
-        and not item.get_closest_marker("flaky_jail")
-    ):
-        if not item.config.getoption("--no-fast-tests"):
+    else:
+        if item.get_closest_marker("core_test"):
+            if not item.config.getoption("--core-tests"):
+                raise pytest.skip.Exception(
+                    "Core tests are disabled, pass '--core-tests' to enable them.",
+                    _use_item_location=True,
+                )
+        if item.get_closest_marker("slow_test"):
+            if not item.config.getoption("--slow-tests"):
+                raise pytest.skip.Exception(
+                    "Slow tests are disabled, pass '--run-slow' to enable them.",
+                    _use_item_location=True,
+                )
+        if test_group_count == 0 and not item.config.getoption("--no-fast-tests"):
```

Comment:
```suggestion
        if test_group_count == 0 and item.config.getoption("--no-fast-tests"):
```
This looks like the right logic if we're respecting the skip message of `Fast tests have been disabled by '--no-fast-tests'`